mirascope/gemini/calls.py

```diff
@@ -0,0 +1,167 @@
+"""A module for calling Google's Gemini Chat API."""
+import datetime
+import logging
+from typing import (
+    Any,
+    AsyncGenerator,
+    ClassVar,
+    Generator,
+)
+
+from google.generativeai import GenerativeModel  # type: ignore
+from google.generativeai.types import ContentsType  # type: ignore
+
+from ..base import BaseCall
+from .tools import GeminiTool
+from .types import (
+    GeminiCallParams,
+    GeminiCallResponse,
+    GeminiCallResponseChunk,
+)
+
+logger = logging.getLogger("mirascope")
+
+
+class GeminiCall(BaseCall[GeminiCallResponse, GeminiCallResponseChunk, GeminiTool]):
+    '''A class for prompting Google's Gemini Chat API.
+
+    This prompt supports the message types: USER, MODEL, TOOL
+
+    Example:
+
+    ```python
+    from google.generativeai import configure  # type: ignore
+    from mirascope.gemini import GeminiPrompt
+
+    configure(api_key="YOUR_API_KEY")
+
+
+    class BookRecommender(GeminiPrompt):
+        prompt_template = """
+        USER: You're the world's greatest librarian.
+        MODEL: Ok, I understand I'm the world's greatest librarian. How can I help?
+        USER: Please recommend some {genre} books.
+
+        genre: str
+
+
+    response = BookRecommender(genre="fantasy").call()
+    print(response.call())
+    #> As the world's greatest librarian, I am delighted to recommend...
+    ```
+    '''
+
+    call_params: ClassVar[GeminiCallParams] = GeminiCallParams(
+        model="gemini-1.0-pro",
```

Comment:
nit: make models into constants default_gemini_model, default_openai_model, etc.

Reply:
I've refactored to single source in types as the default instead.